src/main/scala/org/broadinstitute/hail/driver/AnnotateGlobalExpr.scala

```diff
@@ -60,25 +52,81 @@ object AnnotateGlobalExpr extends Command {
 
     val inserters = inserterBuilder.result()
 
-    if (aggECV.aggregationFunctions.nonEmpty) {
-      val (zVal, seqOp, combOp, resOp) = Aggregators.makeFunctions(aggECV)
+    val (zVal, seqOp, combOp, resOp) = Aggregators.makeFunctions[(Variant, Annotation)](ec, { case (ec, (v, va)) =>
+      ec.setAll(localGlobalAnnotation, v, va)
+    })
 
-      val result = vds.variantsAndAnnotations
-        .treeAggregate(zVal)(seqOp, combOp, depth = HailConfiguration.treeAggDepth(vds.nPartitions))
-      resOp(result)
-    }
+    val result = vds.variantsAndAnnotations
+      .treeAggregate(zVal)(seqOp, combOp, depth = HailConfiguration.treeAggDepth(vds.nPartitions))
+    resOp(result)
+
+    ec.setAll(localGlobalAnnotation)
+    val ga = inserters
+      .zip(fns.map(_ ()))
+      .foldLeft(vds.globalAnnotation) { case (a, (ins, res)) =>
+        ins(a, Option(res))
```

Comment:
`ins` is an Inserter.  It takes `Option[Any]`.  Outside expr, we normally convert `Annotation` values to `Options`.